### ISSUE TITLE
best practice for sending from a routing alias

### ISSUE THREAD

**Issue by ebblake** [ScottPeterJohnson/purelymail-issues#191]
I've just switched to purelymail.com, and it is mostly doing what I need.  However, I just ran into a situation that previously worked with my previous MX host (gandi.net) that failed under purelymail.

Back-story: since I administer my own domain, I love that I can create as many aliases as I want.  More importantly, I have distinct mailboxes; including one for mail to me, and one for mail to my wife.  So one of my first tasks was to tie my own domain to my purelymail account (names changed below to prevent spam), and create two mailboxes: me@example.com, wife@example.com.  Then I created a routing rule that distributes mail sent to both@example.com to redirect to 'me@example.com,wife@example.com'.  So far, so good - this matches what gandi.net let me do by setting up the 'both' alias tied to both the 'me' and the 'wife' mailbox - incoming mail to that address is copied to both our inboxes, where we can each mark the mail read on our own terms.  But when I then tried to send an email claiming to be 'From: both@example.com', I got an error from smtp.purelymail.com stating "SMTP session failed: 530 5.7.1 You (me@example.com) are not authorized to send mail as both@example.com".  This was something I used to be able to do without any effort with my previous MX (although it could conceivably have been a case of my SPF rules being too lax - and I have no idea if I could have sent mail as wife@example.com under their old setup).

I'm guessing I could set up a dummy 'both@example.com' inbox plus a per-box sieve filter that redirects all incoming mail to split out to both intended recipient boxes, as well as configure neomutt so that when the intended From: alias is both@, then log in to SMTP under the 'both@example.com' username rather than the 'me@example.com' username - but I didn't find anything in the docs mentioning that.  And that doesn't scale - while it would let me choose which alias I use in my outgoing mails, my wife can't easily send mails from the same alias without duplicating the dual-SMTP login setup on her preferred email setup, as well as setting up a password into the both@example.com account that we can both share.

Even better, it would be nice if the validation for what From: lines I'm able to use in outgoing mail could look at existing routing rules, and/or if the account admin could list permitted sending aliases associated with each mailbox.  Since I already have a routing rule in place, it would be awesome if, while logged in as 'me@example.com', I can send mail as either 'me@example.com' or as 'both@example.com' (either because me@example.com is included as one of the recipients in the both@example.com routing entry, or because I added 'both@example.com' as a permitted sender alias to the 'me@example.com' mailbox), but NOT send mail as 'wife@example.com'.  Likewise, when my wife logs in as 'wife@example.com', it would be awesome if she can send mail as 'wife@example.com' or 'both@example.com', but not as 'me@example.com'.  That would be a code change rather than a best-practice documentation change, so I can understand if it is harder to implement, but I hope I'm presenting the case well enough.

**Comment by terba** [comment]
TLDR fully, but I think you can set up that at Account -> Sending Address Restrictions. You don't need a real user named _both_, just your two normal users and add two records beside the _Default Purelymail Rules_, one for each real user with _Matching user name_ **both** and select your example.com domain in the lower dropdown. This will let both users send email in their names as usual and also with a From header both@example.com. Should look like this:
![example](https://github.com/ScottPeterJohnson/purelymail-issues/assets/19430601/c7edc2d1-4eeb-4bab-9180-df49cb25524b)
(It's a bit misleading as the UI says _Matching user name_ but in this case _both_ is not a user.)

**Comment by ebblake** [comment]
That looks like it does what I wanted.  I was looking under Users, not Account, which explains why I couldn't find it.  It may still be worth adding better documentation, but since the feature appears to work, I'm also okay if this gets closed.

**Comment by ScottPeterJohnson** [comment]
Sorry for the late reply (I'm extra slow at Github issues it seems).
As terba pointed out, controlling who can send as what is what the Send As is for. It is a bit awkwardly under the Account section, I might have to rethink how it's laid out.

Usually though you have to enable Send As restrictions before the system will ever give you a "You are not authorized to send as" error for an address you own on any user account you own, so _probably_ you enabled it at some point and forgot or didn't realize what it did?

> (It's a bit misleading as the UI says Matching user name but in this case both is not a user.)

The verbiage is pretty unsatisfying. In "address@domain", address is usually the "local part"- but if I put that down, nobody would know what it meant. But "user" can be misunderstood as an actual user too. In this case I guess it can just say "Matching"...

**Comment by ebblake** [comment]
I think I realized what it did when I enabled it as part of first creating my account (I very much don't want my wife spoofing my email address, and I'm sure she doesn't want me spoofing hers), but I failed to realize the knock-on consequences (as long as aliases aren't automatically supported by the default rules, then I need to manually add additional rules duplicating what aliases I've installed, so that we can both send as 'both@example.com').  Changing the UI to say "Matching..." would make sense to me.

**Comment by femme-theory23** [comment]
Closing this as it is more a support ticket than an issue.
